cockatrice/src/player.cpp

```diff
@@ -300,6 +303,57 @@ Player::Player(const ServerInfo_User &info, int _id, bool _local, TabGame *_pare
     rfgMenu->addAction(aViewRfg);
     rfg->setMenu(rfgMenu, aViewRfg);
 
+    {
+        sharedMenu = playerMenu->addMenu(QString());
+
+        sharedMenu->addAction(tr("Draw card from shared"), [=]() {
+            Command_MoveCard cmd;
+            cmd.set_start_zone("shared");
+            cmd.set_start_player_id(this->getId());
+            cmd.set_target_zone("hand");
+            cmd.mutable_cards_to_move()->add_card()->set_card_id(0);
+            sendGameCommand(cmd);
+        });
+
+        sharedMenu->addAction(tr("View shared"),
+                              [=]() { static_cast<GameScene *>(scene())->toggleZoneView(this, "shared", -1); });
+
+        sharedMenu->addAction(tr("Move top shared card to table"), [=]() {
+            Command_MoveCard cmd;
+            cmd.set_start_zone("shared");
+            cmd.set_start_player_id(this->getId());
+            cmd.set_target_zone("table");
+            cmd.mutable_cards_to_move()->add_card()->set_card_id(0);
+            sendGameCommand(cmd);
+        });
+        sharedMenu->addAction(tr("Move X card from your library to shared"), [=]() {
+            bool ok;
+            int number = QInputDialog::getInt(nullptr, tr("Place card X cards from library to shared"),
+                                              tr("How many cards from the top of the deck should this card be placed:"),
```

Comment:
This message is a bit misleading, something like "Amount of cards to move from the top of the deck to shared library:" would be clearer.